### ISSUE TITLE
Attributes setting

### ISSUE THREAD

**Issue by pheuzoune** [StyraHem/ShellyForHASS#165]
Hi, 

Every sensor generated by the custom component has the full list of attributes of the device itself. 

The consequence is that every time an attribute is updated, it triggers an Home Assistant event with the new full list of attrbutes. This list beeing slightly different, it triggers the full process of events handling, meaning : 
 - Storing the event in database
 - Storing the new state in the database
 - Pushing the event to websocket

As an example, I have the sensor "mqtt" on all my devices that I use for monitoring. The full state of the mqtt sensor is modified every time the device push data with a new uptime (which by definition changes all the time) or a different RSSI (which is very variable). 
This result in storing 1 events and 1 new state per minutes (in average) for a sensor that nearly never changes. 

Is it possible to have an extra option like 

```
shelly:
  sensor_disable_attributes: true
```

which should generate entities without any other attributes than the value. 
Or something like
```
shelly:
  sensor_attributes:
    - ssid
    - shelly_type
```

should save many I/Os, many websocket events and many storage for the database.

**Comment by hakana** [comment]
Yes, I'm planning to do an attributes setting

**Comment by hakana** [comment]
Fixed, will be in next beta release

**Comment by medaemon** [comment]
Thank you, I recently add 11 Shelly plug S and it causes heavy IOs and DB size increase...will try it as soon as it will be available :)

**Comment by vistalba** [comment]
Facing into the same issue as well today. I just installed about 20 shellies last week and my DB is now huge (5,7GB). I think the reason is that history is tracked over all this and saved to DB as well.

@hakana Any ETA for new release?

**Comment by hakana** [comment]
There is a beta 9 released yesterday, there you can turn off attributes. I think that it is attributes like rssi, consumtion, uptime that cause traffic.

**Comment by vistalba** [comment]
> There is a beta 9 released yesterday, there you can turn off attributes. I think that it is attributes like rssi, consumtion, uptime that cause traffic.

I just updated to test it out. 
There is an issue. My current consumption only get on value after restart HASS. This value doesn‘t change anymore. If I look at shelly itself it has changes.

Device: shelly 1pm

**Comment by chemelli74** [comment]
> Edit: Just downgraded to 0.1.5 and it is working fine again.

Do you mind test 0.1.6b8 ? Just to be sure the issue is in b9.
Thank you,

Simone

**Comment by vistalba** [comment]
0.1.6b8 looks like it works. 

For me it looks like value does change slower... but it changes.

**Comment by pheuzoune** [comment]
Hello, 

Just tested out the attribute part. 
It's far better, thanks a lot. 

i'll see tomorrow if there's other impact as described on consumption of other.

**Comment by vistalba** [comment]
@chemelli74 / @pheuzoune 

Let me know if you need some more information about config or when I should test other things.

**Comment by hakana** [comment]
Is there on different Shelly devices? For me, it working as expected with consumption.

**Comment by vistalba** [comment]
I have just one Shelly 1pm installed.

My configuation.yaml looks like:
```
shelly:
  discovery: true
  username: "user"
  password: "passwd"
  show_id_in_name: false
  igmp_fix: true
  additional_information: false
  sensors:
  - power
  - battery
  - switch
  - over_temp
  - over_power
  - consumption
  attributes:
  - default
  devices:
    - id: "C4A998"
      light_switch: false
      name: "Heizung"
```

**Comment by hakana** [comment]
Any errors in log?

Try add the igmp_fix: True under shelly: and restart HA

**Comment by vistalba** [comment]
> Any errors in log?
> 
> Try add the igmp_fix: True under shelly: and restart HA

As you can see, it is already there.
No errors in log :-(

After restart the value is NaN and then gets updated but after that it stuck on that value.

**Comment by chemelli74** [comment]
Thread was about "attribute settings"; and it's fixed in 0.1.6 final release.
You can use:

`attributes: []`

to disable all of them.

I proceed to close this issue,

Simone
